src/main/java/com/limechain/trie/structure/decoded/node/DecodedNode.java

```diff
@@ -228,4 +240,221 @@ public byte[] calculateMerkleValue(Function<byte[], byte[]> hashFunction, boolea
 
         return nodeValue;
     }
+
+    /***
+     * Decodes a node value found in a proof into its components.
+     * This can decode nodes no matter their version or hash algorithm.
+     *
+     * @param encoded The byte array representing the encoded node.
+     * @return DecodedNode<Nibbles, List < Byte>> The decoded node with its partial key, children, and storage value.
```

Comment:
[Possible nitpick]
Wrap those generics in a {@code ... } to avoid the HTML trouble with the angle brackets in the JavaDoc :D

```suggestion
     * @return {@code DecodedNode<Nibbles, List <Byte>>} The decoded node...
```

Although I generally stray away from repeating the return type in the JavaDoc, it's kinda visible from the signature.